packages/server/app/Controllers/Http/SearchUsersController.ts

```diff
@@ -6,13 +6,43 @@ import {
     GetUserProfileInformationResponseBody,
     SearchUsersRequestBody,
     SearchUsersResponseBody,
+    UserSettingVisibility,
 } from '@musicroom/types';
 import ForbiddenException from 'App/Exceptions/ForbiddenException';
 import User from 'App/Models/User';
 import invariant from 'tiny-invariant';
 
 const SEARCH_USERS_LIMIT = 10;
 
+function getCounterValueDependingVisibility({
+    fieldValue,
+    fieldVisibility,
+    requestingUserIsfollowingRelatedUser,
+}: {
+    requestingUserIsfollowingRelatedUser: boolean;
+    fieldValue: number;
+    fieldVisibility: UserSettingVisibility;
+}): number | undefined {
+    switch (fieldVisibility) {
+        case UserSettingVisibility.Values.PRIVATE: {
+            return undefined;
+        }
+        case UserSettingVisibility.Values.PUBLIC: {
+            return fieldValue;
+        }
+        case UserSettingVisibility.Values.FOLLOWERS_ONLY: {
+            if (requestingUserIsfollowingRelatedUser) {
+                return fieldValue;
+            }
+            return undefined;
+        }
+        default: {
+            console.error(`unknown switch ue case ${fieldVisibility}`);
+            return undefined;
```

Comment:
I like to throw an error when an unreachable state is reached, because it means the program can not continue to work properly. Therefore, the best thing to do is to stop it as soon as possible 😆